### PR TITLE
Allow examples and notes to be cited

### LINKED ISSUE #148: Reference linking for `Note` and `Example`
Need to add `id` to `<p class="note">` and `<p class="example">` to account for referencing from then text (optionally). 

As per ISO directive 2:
```
24.4 Referencing
When notes are referred to, use, for example, the following forms for references:

• “an explanation is provided in 7.1, Note 2”;

• “see 8.6, Note 3”.
```
and 
```
25.4 Referencing
When examples are referred to, use for example, the following forms for references:

• “see 6.6.3, Example 5”;

• “Clause 4, Example 2 lists …”
```

### PR DESCRIPTION
**Pull request by palemieux**
Closes #148

### CODE REVIEW

File: doc/main.html [hunk 1]
```diff
@@ -937,43 +937,57 @@ <h4>Internal code snippets</h4>
...
+        </div>
+
       </section>
 
       <section id="sec-example-class">
         <h4>Examples</h4>
 
-        <p>Examples can be inserted by using <code>div</code> element whose <code>class</code> attribute is equal to
-        <code>example</code>.</p>
+        <p>An example is inserted by using <code>div</code> element whose <code>class</code> attribute is equal to
```
Comment (SteveLLamb) [inline doc/main.html:971]:
```suggestion
        <p>An example is inserted by using a <code>p</code> or <code>div</code> element whose <code>class</code> attribute is equal to
```

Comment (SteveLLamb) [reply doc/main.html:971]: This still applies, yes?

Comment (palemieux) [reply doc/main.html:971]: Hadn't we decided that examples could only be `div`?

Comment (SteveLLamb) [reply doc/main.html:971]: I though we expanded for `div`, not disallowed `p`. Both currently work. I'm fine with it either way tbh.

Comment (palemieux) [reply doc/main.html:971]: Ok. Let me add tests for both.

**Review by SteveLLamb — APPROVED** [review]

**Pull request merged**